stripe/src/main/java/com/stripe/android/util/StripeNetworkUtils.java

```diff
@@ -37,13 +41,41 @@
         cardParams.put("address_country", StripeTextUtils.nullIfBlank(card.getAddressCountry()));
 
         // Remove all null values; they cause validation errors
-        for (String key : new HashSet<>(cardParams.keySet())) {
-            if (cardParams.get(key) == null) {
-                cardParams.remove(key);
-            }
-        }
+        removeNullParams(cardParams);
 
         tokenParams.put("card", cardParams);
         return tokenParams;
     }
+
+    @NonNull
+    public static Map<String, Object> hashMapFromBankAccount(@NonNull BankAccount bankAccount) {
+        Map<String, Object> tokenParams = new HashMap<>();
+        Map<String, Object> accountParams = new HashMap<>();
+
+        accountParams.put("country", bankAccount.getCountryCode());
+        accountParams.put("currency", bankAccount.getCurrency());
+        accountParams.put("account_number", bankAccount.getAccountNumber());
+        accountParams.put("routing_number",
+                StripeTextUtils.nullIfBlank(bankAccount.getRoutingNumber()));
+        accountParams.put("account_holder_name",
+                StripeTextUtils.nullIfBlank(bankAccount.getAccountHolderName()));
+        accountParams.put("account_holder_type",
+                StripeTextUtils.nullIfBlank(bankAccount.getAccountHolderType()));
+
+        // Remove all null values; they cause validation errors
+        removeNullParams(accountParams);
+
+        tokenParams.put("bank_account", accountParams);
+        return tokenParams;
+    }
+
+    static void removeNullParams(Map<String, Object> mapToEdit) {
+        // Remove all null values; they cause validation errors
+        for (String key : new HashSet<>(mapToEdit.keySet())) {
```

Comment:
Yes! The `ConcurrentModificationException` that we get when we try to remove an element while using a for-each loop happens because the for-each loop internally uses an `Iterator` that we don't have access to and while iterating we can only use the `Iterator` to remove the element.

The `removeIf` was added as part of Java8 and it internally uses an `Iterator` to remove the elements: http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/8u40-b25/java/util/Collection.java#Collection.removeIf%28java.util.function.Predicate%29